src/components/GridContainer/GridContainer.tsx

```diff
@@ -0,0 +1,129 @@
+  | "space-evenly"
+  | "start"
+  | "stretch"
+  | "end"
+  | "left"
+  | "right";
+
+export interface GridContainerProps extends HTMLAttributes<HTMLDivElement> {
+  alignItems?: ItemsOptions;
+  alignContent?: ContentOptions;
+  children?: React.ReactNode;
+  display?: DisplayOptions;
+  columnGap?: GapOptions;
+  gap?: GapOptions;
+  gridAutoColumns?: string;
+  gridAutoFlow?: FlowOptions;
+  gridAutoRows?: string;
+  gridTemplateAreas?: string;
+  gridTemplateColumns?: string;
+  gridTemplateRows?: string;
+  gridTemplate?: string;
+  isResponsive?: boolean;
+  justifyContent?: ContentOptions;
+  justifyItems?: ItemsOptions;
+  rowGap?: GapOptions;
+}
+
+const GridContainer = ({
+  alignItems = "stretch",
+  alignContent = "stretch",
+  children,
+  columnGap = "none",
+  display = "grid",
+  gap = "unset",
+  gridAutoColumns = "",
+  gridAutoFlow = "row",
+  gridAutoRows = "",
+  gridTemplateAreas = "",
+  gridTemplateColumns = "",
+  gridTemplateRows = "",
+  gridTemplate = "",
+  isResponsive = true,
+  justifyContent = "stretch",
+  justifyItems = "stretch",
+  rowGap = "none",
+  ...props
+}: GridContainerProps) => {
+  return (
+    <Wrapper
+      $alignItems={alignItems}
+      $alignContent={alignContent}
+      $columnGap={columnGap}
+      $display={display}
+      $gap={gap}
+      $gridAutoColumns={gridAutoColumns}
+      $gridAutoFlow={gridAutoFlow}
+      $gridAutoRows={gridAutoRows}
+      $gridTemplateAreas={gridTemplateAreas}
+      $gridTemplateColumns={gridTemplateColumns}
+      $gridTemplateRows={gridTemplateRows}
+      $gridTemplate={gridTemplate}
+      $isResponsive={isResponsive}
+      $justifyContent={justifyContent}
+      $justifyItems={justifyItems}
+      $rowGap={rowGap}
+      data-testid="grid-container"
+      {...props}
+    >
+      {children}
+    </Wrapper>
+  );
+};
+
+const Wrapper = styled.div<{
+  $alignContent: ContentOptions;
+  $alignItems: ItemsOptions;
+  $columnGap: GapOptions;
+  $display: DisplayOptions;
+  $gap: GapOptions;
+  $gridAutoColumns: string;
+  $gridAutoFlow: FlowOptions;
+  $gridAutoRows: string;
+  $gridTemplateAreas: string;
+  $gridTemplateColumns: string;
+  $gridTemplateRows: string;
+  $gridTemplate: string;
+  $isResponsive: boolean;
+  $justifyContent: ContentOptions;
+  $justifyItems: ItemsOptions;
+  $rowGap: GapOptions;
+}>`
+  align-items: ${({ $alignItems = "stretch" }) => $alignItems};
+  align-content: ${({ $alignContent = "stretch" }) => $alignContent};
+  column-gap: ${({ theme, $columnGap }) => theme.click.gridContainer.gap[$columnGap]};
+  display: ${({ $display }) => $display};
+  gap: ${({ theme, $gap }) => theme.click.gridContainer.gap[$gap]};
+  grid-auto-columns: ${({ $gridAutoColumns }) => $gridAutoColumns || "auto"};
+  grid-auto-flow: ${({ $gridAutoFlow }) => $gridAutoFlow || "auto"};
+  grid-auto-rows: ${({ $gridAutoRows }) => $gridAutoRows || "auto"};
+  grid-template-area: ${({ $gridTemplateAreas }) => $gridTemplateAreas || "auto"};
```

Comment:
I think some of the properties become not working with the others
For example the area with the columns and rows don't match
May be we should do a conditional